pkg/cloud/vsphere/services/govmomi/util.go

```diff
@@ -95,22 +95,32 @@ func getNetworkStatus(ctx *context.MachineContext) ([]v1alpha1.NetworkStatus, er
 	}
 	ctx.Logger.V(6).Info("got allNetStatus", "status", allNetStatus)
 	apiNetStatus := []v1alpha1.NetworkStatus{}
-	for _, netDevSpec := range ctx.MachineConfig.MachineSpec.Network.Devices {
-		for _, netStatus := range allNetStatus {
-			if netDevSpec.UUID == netStatus.UUID {
-				apiNetStatus = append(apiNetStatus, v1alpha1.NetworkStatus{
-					Connected:   netStatus.Connected,
-					UUID:        netStatus.UUID,
-					IPAddrs:     netStatus.IPAddrs,
-					MACAddr:     netStatus.MACAddr,
-					NetworkName: netStatus.NetworkName,
-				})
-			}
-		}
+	for _, s := range allNetStatus {
+		apiNetStatus = append(apiNetStatus, v1alpha1.NetworkStatus{
+			Connected:   s.Connected,
+			IPAddrs:     sanitizeIPAddrs(ctx, s.IPAddrs),
+			MACAddr:     s.MACAddr,
+			NetworkName: s.NetworkName,
+		})
 	}
 	return apiNetStatus, nil
 }
 
+func sanitizeIPAddrs(ctx *context.MachineContext, ipAddrs []string) []string {
+	if len(ipAddrs) == 0 {
+		return nil
+	}
+	newIPAddrs := []string{}
+	for _, i := range ipAddrs {
```

Comment:
s/i/ipAddr

Reply:
Hi @andrewsykim,

I used `i` instead of `ipAddr` because of the name of the parameter. It's a short function, and I don't think there's much room for confusion. Again, if I update code for some reason I'll change it to `addr`.

Reply:
Hi @andrewsykim,

Updated.